### ISSUE TITLE
Not working at all

### ISSUE THREAD

**Issue by MuTsunTsai** [DennisvHest/vscode-npm-browser#33]
Recently it starts to show complete gibberish on screen. I've included a screenshot with error messages in developer console, hope this helps.

![image](https://user-images.githubusercontent.com/39870664/152635934-f8cc6a53-c2ff-46c9-b6a0-351d5b9f244c.png)

**Comment by DennisvHest** [comment]
Hey @MuTsunTsai, thanks for the bug report.

I have seen multiple people report this issue, yet I have not been able to reproduce it. I see that content security policies are preventing scripts from being loaded. I see vue.js and shrewd.js are blocked, but these aren't scripts from my extension. Maybe other extensions are injecting scripts into web views?

I have created a build of the extension that does not enforce content security policies. It would really help me if you could install this build and check if the issue is fixed.

You can find the release here: https://github.com/DennisvHest/vscode-npm-browser/releases/tag/1.1.1-fix You can download npm-browser-1.1.1.vsix and then install it via the extensions tab in VSCode from here (uninstall the current NPM browser extension first):

![image](https://user-images.githubusercontent.com/18610083/152638228-0c6eaa74-9a1b-484f-9f6c-8c10dfbdb348.png)

**Comment by MuTsunTsai** [comment]
I've tried the build, and it's still not working:

![image](https://user-images.githubusercontent.com/39870664/152642477-26264dcc-bcd3-46ea-a5b8-77df1afcef79.png)

In this particular case, `app.js` and `shrewd.js` are files in my project folder. As for `vue.js`, it only exists in `node_modules`. I'm not sure why any of them is requested by the extension. The next screenshot show where in `main.js` did the request go wrong.

![image](https://user-images.githubusercontent.com/39870664/152642708-589e1d7f-5c6b-44e8-95bd-863dfe19e68e.png)

However, I'm more concerning with the error where it says `vscode is not defined`, since that part comes directly from your extension. This is where that error came from:

![image](https://user-images.githubusercontent.com/39870664/152642788-9a97dea1-8582-4768-9e72-3c119520a9b2.png)

One more thing: the gibberish text displayed is mostly from `README.md` of my project.

**Comment by DennisvHest** [comment]
Sorry to hear that it is not working without the content security policies, but I have cloned your shrewd repository and I am now able to reproduce the error. When I have time, I will take a closer look to figure out the issue.

It is weird that vscode is undefined, because that is a global object that VSCode inserts into web views. That's why I think something weird is happening before that line of code.

Thanks for the further info!

**Comment by MuTsunTsai** [comment]
Really appreciate it. By the way, I'm pretty sure that your extension used to work with my Shrewd repo (maybe a week or so ago), and I think neither of the repo or the extension had changed significantly since, so I'm more suspecting it has to do with recent VS Code updates. Just a thought.

**Comment by DennisvHest** [comment]
I have fixed the issue and created a new release ([1.1.2](https://github.com/DennisvHest/vscode-npm-browser/releases/tag/1.1.2)).

It wasn't an issue with VSCode. If a closing script tag was present in the package.json, the rest of the readme contents would be written to the webview. The package.json contents are inserted into the webview as a json object, but because of how HTML parsing works, the script tag is prematurely closed (even though it is in a json string). I have fixed this by HTML escaping the readme and description contents of the package.json.
